src/datasets/arrow_reader.py

```diff
@@ -43,7 +43,7 @@
 _SUB_SPEC_RE = re.compile(
     r"""
 ^
- (?P<split>\w+)
+ (?P<split>\w+(-\w+)*)
```

Comment:
@lhoestq - I need this to allow splits like `test-de`. Is that fine for you? 
The regex is simply extended to allow for split names that have 0-n hyphens.

Reply:
@patrickvonplaten I forgot I had the same problem with ELI5. switching to `test_de` should work.

@lhoestq I'm also not a fan of split names with intrinsic meanings. Where is the regular expression documented currently?